presentation/frontend/src/test/StreamedResponseGenerator.test.ts

```diff
@@ -0,0 +1,225 @@
+import { StreamedResponseGenerator, StreamedResponseGeneratorProps } from "../presentation-frontend/StreamedResponseGenerator";
+import { expect } from "chai";
+import sinon from "sinon";
+import { PagedResponse, PageOptions } from "@itwin/presentation-common";
+import { ResolvablePromise } from "@itwin/presentation-common/lib/cjs/test";
+
+describe("StreamedResponseGenerator", () => {
+  it("should run requests concurrently", async () => {
+    const total = 10;
+    const firstBatchPromise = new ResolvablePromise<PagedResponse<number>>();
+    const restBatchesPromise = new ResolvablePromise<PagedResponse<number>>();
+    const fakeGetBatch = sinon.fake(async (_, idx: number) => {
+      const result = idx ? restBatchesPromise : firstBatchPromise;
+      return result as unknown as Promise<PagedResponse<number>>;
```

Comment:
is `as unknown as Promise<PagedResponse<number>>` necessary?

Reply:
```
Type 'ResolvablePromise<PagedResponse<number>>' is missing the following properties from type 'Promise<PagedResponse<number>>': catch, finally, [Symbol.toStringTag]
```

Reply:
Let's fix it like this: https://github.com/iTwin/presentation/blob/350aef3ec6f283c2d902096e343fc6a52a14a464/packages/components/src/test/_helpers/Promises.ts#L31

Reply:
Done